openedx/core/djangoapps/programs/utils.py

```diff
@@ -285,10 +285,12 @@ def completed_programs_with_available_dates(self):
         Returns a dict of {uuid_string: available_datetime}
         """
         completed = {}
+        programs_without_certificates = configuration_helpers.get_value('programs_without_certificates', [])
         for program in self.programs:
-            available_date = self._available_date_for_program(program)
-            if available_date:
-                completed[program['uuid']] = available_date
+            if program['uuid'] not in programs_without_certificates:
```

Comment:
This check should be in the place where we are awarding certificates for program instead of  in `completed_programs_with_available_dates `  method .